src/tool_formparse.c

```diff
@@ -381,8 +384,18 @@ static char *get_param_word(char **str, char **end_pos, char endchar)
           while(ptr < *end_pos);
           *end_pos = ptr2;
         }
-        while(*ptr && *ptr != ';' && *ptr != endchar)
+        ++ptr;
```

Comment:
Why the extra increment?

Reply:
To advance past the closing `"`, which would otherwise trigger the `ISSPACE` check.

Reply:
> Why the extra increment?

I'm also curious about this. When the string is unescaped then can ptr be advanced past *end_pos in the escaped loop (passed the pointer to the end `"`). Also could *ptr == endchar. In either case should we be incrementing it

Reply:
I *think* `ptr == end_pos` is guaranteed, since the final `"` wasn't escaped and so can't be skipped past. (If it was escaped then it would be considered missing and we wouldn't end up here.)

But to make it more robust, maybe I could do
```suggestion
        ptr = end_pos + 1;
```
or
```suggestion
        if (*ptr == '"')
          ++ptr;
```
or down below:
```c
          if(!ISSPACE(*ptr) && *ptr != '"')
```
(though that last one would prevent some reasonable warnings)

Reply:
> I _think_ `ptr == end_pos` is guaranteed, since the final `"` wasn't escaped and so can't be skipped past. (If it was escaped then it would be considered missing and we wouldn't end up here.)

Looks like you're right but AFAICT it's still possible that a double quote could be used as endchar.
```suggestion
        if(*ptr != endchar)
          ++ptr;
```

Reply:
In practice `endchar` is always either `\0` or `,` so it shouldn't matter and changing it isn't a compatibility hazard.

Even if `endchar` were `"` I don't think you'd want the closing quote to do double duty as a separator, so advancing past it unconditionally makes more sense than the old behavior.

(The purpose of `endchar` is to separate files in an argument like `-Fx=@foo.txt;filename="foo",bar.txt;filename="bar"`. If it's `\0` it's effectively ignored and it's only `,` for the `=@` case.)

Reply:
A double quote may not make sense as the endchar currently but if it could happen I'd rather have the same behavior not to advance ptr if *ptr is endchar.

Reply:
Why? The new behavior IMO makes more sense.

The intent of the loop is to move past the end of the value to a delimiter that's after/between fields. If in `-Fx=@foo.txt;filename="foo",bar.txt;filename=bar` you replaced the `,` by a `"` and parsed it with an `endchar` of `"` it'd work with the new behavior but raise a warning with the old behavior. I think you'd want each `"` character to be used as *either* a quote or an endchar, but not both at the same time.

Reply:
The loops in the function will not advance ptr when *ptr == endchar, so I prefer it for consistency. I'm for retaining the existing behavior even if what we are discussing is likely improbable.

Reply:
Isn't it currently impossible rather than improbable? It's a static function, only called by another static function which is only ever called with a char literal. `endchar` is also never assigned to or referenced.

If you don't think that's enough I'll defer to your judgment.